### PR TITLE
Fixes for dimoc 338

### PR DESCRIPTION
**Pull request by rjzondervan**
- Limit
- Extend multiple
- Extend all

### CODE REVIEW

**Review by bbrands02 — APPROVED** [review]

File: lib/Service/ObjectService.php [hunk 1]
```diff
@@ -328,6 +328,11 @@ public function getResultArrayForRequest(string $objectType, array $requestParam
 		$offset = $requestParams['offset'] ?? $requestParams['_offset'] ?? null;
 		$order = $requestParams['order'] ?? $requestParams['_order'] ?? null;
 		$extend = $requestParams['extend'] ?? $requestParams['_extend'] ?? null;
+		$page = $requestParams['page'] ?? $requestParams['_page'] ?? null;
+
+		if($page !== null && isset($limit)) {
```
Comment (WilcoLouwerse) [inline lib/Service/ObjectService.php:333]: spacing inconsistency, i would do:
if (
not
if(

as we always have

File: lib/Service/ObjectService.php [hunk 2]
```diff
@@ -372,9 +377,15 @@ public function getResultArrayForRequest(string $objectType, array $requestParam
 	 */
 	public function extendEntity(mixed $entity, array $extend): array
 	{
+		$surpressMapperError = false;
 		// Convert the entity to an array if it's not already one
 		$result = is_array($entity) ? $entity : $entity->jsonSerialize();
 
+		if(in_array(needle: 'all', haystack: $extend) === true) {
```
Comment (WilcoLouwerse) [inline lib/Service/ObjectService.php:384]: spacing inconsistency, i would do:
if (
not
if(

as we always have

File: lib/Service/ObjectService.php [hunk 3]
```diff
@@ -403,6 +414,9 @@ public function extendEntity(mixed $entity, array $extend): array
 					$propertyObject = $singularProperty;
 				} catch (Exception $e) {
 					// If still no mapper, throw a no mapper available error
+					if($surpressMapperError === true) {
```
Comment (WilcoLouwerse) [inline lib/Service/ObjectService.php:417]: spacing inconsistency, i would do:
if (
not
if(

as we always have

**Review by WilcoLouwerse — APPROVED** [review]
I would love it if I no longer have to fix spacing inconsistencies

**Review by WilcoLouwerse — APPROVED** [review]

File: lib/Controller/PublicationsController.php [hunk 4]
```diff
@@ -95,8 +95,16 @@ public function index(ObjectService $objectService): JSONResponse
      */
     public function show(string|int $id, ObjectService $objectService): JSONResponse
     {
+		$parameters = $this->request->getParams();
+
+		$extend = [];
+
+		if(isset($parameters['extend']) === true) {
```
Comment (WilcoLouwerse) [inline lib/Controller/PublicationsController.php:102]: if spacing

File: lib/Controller/SearchController.php [hunk 5]
```diff
@@ -161,8 +161,16 @@ public function publications(): JSONResponse
 	 */
 	public function publication(string|int $publicationId): JSONResponse
 	{
-		// Get the publication object by ID
-		$object = $this->objectService->getObject('publication', $publicationId);
+		$parameters = $this->request->getParams();
+		
+		$extend = [];
+
+		if(isset($parameters['extend']) === true) {
```
Comment (WilcoLouwerse) [inline lib/Controller/SearchController.php:168]: if spacing

**Review by WilcoLouwerse — APPROVED** [review]

**Review by WilcoLouwerse — APPROVED** [review]

**Review by WilcoLouwerse — APPROVED** [review]

**Pull request merged**